pmxbot/slack.py

```diff
@@ -64,14 +65,19 @@ def get_email_username_map(self, ttl_hash=get_ttl_hash()):
             }
 
     def start(self):
-        res = self.slack.rtm_connect()
-        assert res, "Error connecting"
         self.init_schedule(self.scheduler)
+        threading.Thread(target=self.run_scheduler_loop).start()
+
+        @self.slack.on("message")
+        def handle_payload(client, event):
+            self.handle_message(event)
+
+        self.slack.start()
+
+    def run_scheduler_loop(self):
         while True:
-            for msg in self.slack.rtm_read():
-                self.handle_message(msg)
             self.scheduler.run_pending()
-            time.sleep(0.1)
+            time.sleep(1)
```

Comment:
Why change the sleep time here? Setting to `1` would mean that scheduled events could happen up to 1 sec after they're scheduled, which could potentially create a user-visible delay.

Reply:
My thinking was that 0.1 was a little low and would mean a lot of loop spinning. I think in reality increasing it to 1 would have little noticeable impact. From what I understand any pmxbot actions generated from direct user interaction (e.g. `!thank`) do not get pushed onto the scheduler queue, they're dealt with when the message event is received. This would only be a slight delay in the actions of other plugins (e.g. standup notifications and webhooks) by up to one second.

However, sticking with the philosophy of minimal changes here, I will change back to `0.1`